packages/vite-plugin-cloudflare/src/runner-worker/index.ts

```diff
@@ -329,3 +344,39 @@ export function createDurableObjectWrapper(
 
 	return Wrapper;
 }
+
+export function createWorkflowEntrypointWrapper(
+	className: string,
+): WorkflowEntrypointConstructor<WrapperEnv> {
+	class Wrapper extends WorkflowEntrypoint<WrapperEnv> {}
+
+	for (const key of WORKFLOW_ENTRYPOINT_KEYS) {
+		Wrapper.prototype[key] = async function (...args: unknown[]) {
+			const entryPath = this.env.__VITE_ENTRY_PATH__;
+			const ctor = (await getWorkerEntryExport(
+				entryPath,
+				className,
+			)) as WorkflowEntrypointConstructor;
+			const userEnv = stripInternalEnv(this.env);
+			const instance = new ctor(this.ctx, userEnv);
```

Comment:
sorry I see that we do this for entrypoints and durable objects, I must have missed this before
(and sorry as I feel like this might be a silly question 😓)

could you explain why we create a new instance of the entry export for every key and call? is this correct? don't we end up with a bunch of different instances of ctor?

Reply:
See the comment in the description:

_I initially created a more complex implementation that held onto class instances across requests (similar to the Durable Objects implementation). This turned out to not be necessary because the Workflow create method is a one time trigger and the instance ID is unique per Workflow (https://developers.cloudflare.com/workflows/build/rules-of-workflows/#instance-ids-are-unique). I therefore simplified the implementation._

From my testing, it seems that calling `workflow.create` creates the instance and then, from that point on, the same ID will always refer to the same instance. So this code actually only gets called once per ID.

For Durable Objects, we do hold on to the same instance unless the user code changes. For `WorkerEntrypoint`, I think it is appropriate to create a new instance on each call as they are stateless.

Reply:
Ah I see, yeah I missed the `kInstance` caching/check for DurableObjects

Ok makes sense 🤔 

Thanks for the explanation 😄